src/starknet_client/mod.rs

```diff
@@ -367,7 +367,7 @@ impl<P: Provider + Send + Sync> KakarotClient<P> {
             }
         };
 
-        let chain_id = request.chain_id.unwrap_or_else(|| CHAIN_ID.into());
+        let chain_id = request.chain_id.unwrap();
```

Comment:
No unwraps, we should handle the error, let's error out if the chain_id isn't provided